### PR TITLE
Use EditAuctionViewModel in Admin::AuctionsController

### PR DESCRIPTION
**Pull request by adelevie**
- AuctionEditViewModel -> EditAuctionViewModel
- Set up empty specs to test failed admin auction update.

### CODE REVIEW

File: spec/controllers/admin/auctions_controller_spec.rb [hunk 1]
```diff
@@ -22,4 +22,19 @@
       expect(auction.auction).to eq(auction_record)
     end
   end
+
+  describe '#update' do
```
Comment (adelevie) [inline spec/controllers/admin/auctions_controller_spec.rb]: 👍 !

**Comment by adelevie** [comment]
This PR is ready. Zenhub is futzing with my ability to edit the PR title to remove "WIP".

File: spec/controllers/admin/auctions_controller_spec.rb [hunk 2]
```diff
@@ -16,10 +17,40 @@
...
+    context 'when the params are invalid' do
+      it 'assigns the un-updated auction' do
+        user = create(:admin_user)
+        auction_record = create(:auction)
+
+        update_auction_double = double
+        allow(update_auction_double).to receive(:perform)
```
Comment (jessieay) [inline spec/controllers/admin/auctions_controller_spec.rb]: look at this! mocking and stubbing like a PRO 💯

File: spec/controllers/admin/auctions_controller_spec.rb [hunk 3]
```diff
@@ -16,10 +17,40 @@
...
       auction_record = create(:auction)
       get :show, { id: auction_record.id }, user_id: user.id
       auction = assigns(:auction)
       expect(auction.auction).to eq(auction_record)
     end
   end
+
+  describe '#update' do
+    context 'when the params are invalid' do
+      it 'assigns the un-updated auction' do
```
Comment (jessieay) [inline spec/controllers/admin/auctions_controller_spec.rb]: and renders edit?

File: spec/controllers/admin/auctions_controller_spec.rb [hunk 4]
```diff
@@ -16,10 +17,40 @@
...
+          auction: {
+            title: 'new title'
+          }
+        }
+
+        put :update, params, user_id: user.id
+        auction = assigns(:auction)
+
+        expect(auction).to be_a(Admin::EditAuctionViewModel)
+        expect(auction.record.id).to eq(auction_record.id)
```
Comment (jessieay) [inline spec/controllers/admin/auctions_controller_spec.rb]: thoughts on adding a third expectation:

`expect(response).to render_template(:edit)`

?

Comment (adelevie) [inline spec/controllers/admin/auctions_controller_spec.rb]: ➕

File: spec/controllers/admin/auctions_controller_spec.rb (hunk 3 above)
Comment (adelevie) [inline spec/controllers/admin/auctions_controller_spec.rb]: ➕

File: spec/controllers/admin/auctions_controller_spec.rb [hunk 5]
```diff
@@ -1,10 +1,11 @@
 require 'rails_helper'
 
 describe Admin::AuctionsController do
-  let(:user) { create(:admin_user) }
+  let(:user) {  }
```
Comment (harrisj) [inline spec/controllers/admin/auctions_controller_spec.rb]: Do you still need this `let` declaration?

Comment (adelevie) [inline spec/controllers/admin/auctions_controller_spec.rb]: oh good catch. I'll ⚡ it.

**Pull request merged**